### PR TITLE
[POAE7-2758] Migrate nextgen test cases.

### PR DESCRIPTION
**Pull request by jikunshang**
### What changes were proposed in this pull request?

Migrate nextgen test cases to `CiderNextegenQueryRunner` and `CiderNextegenTestBase` 

### Why are the changes needed?

separate code from origin CiderQueryRunner/RuntimeModule.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

UT

### CODE REVIEW

**Comment by jikunshang** [comment]
@xieqi @yma11 Please take a review, thanks.

File: src/cider/exec/processor/DefaultBatchProcessor.cpp [hunk 1]
```diff
@@ -28,6 +28,25 @@
...
+    case cider::jitlib::ERROR_CODE::ERR_OUT_OF_TIME:
+      return "Query execution has exceeded the time limit";
+    case cider::jitlib::ERROR_CODE::ERR_INTERRUPTED:
+      return "Query execution has been interrupted";
+    case cider::jitlib::ERROR_CODE::ERR_SINGLE_VALUE_FOUND_MULTIPLE_VALUES:
+      return "Multiple distinct values encountered";
+    case cider::jitlib::ERROR_CODE::ERR_WIDTH_BUCKET_INVALID_ARGUMENT:
+      return "Arguments of WIDTH_BUCKET function does not satisfy the condition";
+    default:
+      return "Cider Runtime Other error: code " + std::to_string(error_code);
```
Comment (yma11) [inline src/cider/exec/processor/DefaultBatchProcessor.cpp:46]: `Unknown runtime error with code:` ?

Comment (jikunshang) [reply src/cider/exec/processor/DefaultBatchProcessor.cpp:46]: would be enum id.

File: src/cider/tests/utils/CiderArrowChecker.cpp [hunk 2]
```diff
@@ -43,27 +43,81 @@ bool checkArrowBuffer(const struct ArrowArray* expect_array,
 
   auto expect_null_buffer = reinterpret_cast<const uint8_t*>(expect_array->buffers[0]);
   auto actual_null_buffer = reinterpret_cast<const uint8_t*>(actual_array->buffers[0]);
-  if (expect_null_buffer == nullptr && actual_null_buffer == nullptr) {
+
+  if (expect_null_buffer && actual_null_buffer) {
```
Comment (yma11) [inline src/cider/tests/utils/CiderArrowChecker.cpp:47]: is it possible to detect a scenario and pop message like "values are equal but nulls are different"?

File: src/cider/exec/processor/DefaultBatchProcessor.cpp [hunk 3]
```diff
@@ -44,7 +63,8 @@ DefaultBatchProcessor::DefaultBatchProcessor(const plan::SubstraitPlanPtr& plan,
       std::make_shared<generator::SubstraitToRelAlgExecutionUnit>(plan_->getPlan());
   RelAlgExecutionUnit ra_exe_unit = translator->createRelAlgExecutionUnit();
   jitlib::CompilationOptions co;
-  codegen_context_ = nextgen::compile(ra_exe_unit, co);
+  cider::exec::nextgen::context::CodegenOptions codegen_options{true};
```
[the remark this answers is not in the document]
Comment (yma11) [reply src/cider/exec/processor/DefaultBatchProcessor.cpp:66]: But better to have that, or at least you need to do hard code change when you do benchmarks against w/ or w/o error check.

File: src/cider/tests/utils/CiderNextgenTestBase.cpp [hunk 4]
```diff
@@ -39,19 +39,75 @@ void CiderNextgenTestBase::assertQuery(const std::string& sql,
...
+    struct ArrowSchema output_schema;
+    cider_nextgen_query_runner_->runQueryOneBatch(wrong_sql,
+                                                  *input_array_,
+                                                  *input_schema_,
+                                                  output_array,
+                                                  output_schema,
+                                                  codegen_options_);
+  } catch (const CiderException& e) {
+    LOG(ERROR) << e.what();
+    return true;
```
Comment (kd97) [inline src/cider/tests/utils/CiderNextgenTestBase.cpp:82]: may throw other exception(ciderCompileException), but also pass test here? can return expectTrue(expect_error_code, actual_error_code)?

File: src/cider/exec/nextgen/context/CodegenContext.h [hunk 5]
```diff
@@ -87,6 +88,7 @@ using AggExprsInfoVector = std::vector<AggExprsInfo>;
 
 struct CodegenOptions {
   bool needs_error_check = false;
+  jitlib::CompilationOptions co = jitlib::CompilationOptions{};
 };
```
Comment (kd97) [inline src/cider/exec/nextgen/context/CodegenContext.h:92]: need to move `needs_error_check` into compilationOptions?

Comment (jikunshang) [reply src/cider/exec/nextgen/context/CodegenContext.h:92]: @bigPYJ1151 any suggestion?

Comment (yma11) [reply src/cider/exec/nextgen/context/CodegenContext.h:92]: I remember there is different definition between `CodegenOptions ` and `CompilationOptions` per @bigPYJ1151 ?

File: src/cider/tests/nextgen/functional/CiderMathOpNextgenTest.cpp [hunk 6]
```diff
@@ -56,157 +58,152 @@ class CiderMathOpNullableArrowTest : public CiderTestBase {
 
 TEST_F(CiderMathOpNullableArrowTest, ConstantValueMathOpTest) {
   GTEST_SKIP();
```
Comment (yma11) [inline src/cider/tests/nextgen/functional/CiderMathOpNextgenTest.cpp:60]: Seems there is some unexpected `GTEST_SKIP` in this file, can you help remove them except the `Decimal `one?

**Review by yma11 — DISMISSED** [review]

File: src/cider/tests/utils/CiderNextgenTestBase.cpp (hunk 4 above)
Comment (jikunshang) [reply src/cider/tests/utils/CiderNextgenTestBase.cpp:82]: seems need a lot change. we can refactor this in later PR.

**Review by kd97 — APPROVED** [review]

**Review by yma11 — APPROVED** [review]

**Pull request merged**